nomad/structs/consul.go

```diff
@@ -3,34 +3,102 @@
 
 package structs
 
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+)
+
 // Consul represents optional per-group consul configuration.
 type Consul struct {
 	// Namespace in which to operate in Consul.
+	//
+	// Namespace is set on the client and can be overridden by the consul.namespace
+	// field in the jobspec.
 	Namespace string
+
+	// UseIdentity tells the server to sign identities for Consul. In Nomad 1.9+ this
+	// field will be ignored (and treated as though it were set to true).
+	//
+	// UseIdentity is set on the server.
+	UseIdentity bool
+
+	// ServiceIdentity is intended to reduce overhead for jobspec authors and make
+	// for graceful upgrades without forcing rewrite of all jobspecs. If set, when a
+	// job has a service block with the “consul” provider, the Nomad server will sign
+	// a Workload Identity for that service and add it to the service block. The
+	// client will use this identity rather than the client's Consul token for the
+	// group_service and envoy_bootstrap_hook.
+	//
+	// The name field of the identity is always set to
+	// "consul-service/${service_name}-${service_port}".
+	//
+	// ServiceIdentity is set on the server.
+	ServiceIdentity *WorkloadIdentity
+
+	// TemplateIdentity is intended to reduce overhead for jobspec authors and make
+	// for graceful upgrades without forcing rewrite of all jobspecs. If set, when a
+	// job has both a template block and a consul block, the Nomad server will sign a
+	// Workload Identity for that task. The client will use this identity rather than
+	// the client's Consul token for the template hook.
+	//
+	// The name field of the identity is always set to "consul".
+	//
+	// TemplateIdentity is set on the server.
+	TemplateIdentity *WorkloadIdentity
 }
```

Comment:
Is this the right place for these changes? This struct represents the `group` `consul` block, not the `consul` agent config block, which is here:
https://github.com/hashicorp/nomad/blob/0b9cf4e7b708accb11cf6e16c1b9648ad15f62c2/nomad/structs/config/consul.go#L17-L29